.github/workflows/ci.yml

```diff
@@ -22,8 +26,39 @@ jobs:
     - name: Install Newman dependency
       run: npm install --prefix tests/postman/
 
-    - name: Run the API and Postman's tests
-      run: |
-        cd tests/postman/ && newman run *
+    - name: Start Netlify Dev for local function testing
+      run:  npm run netlify-dev & sleep 10 && cd tests && node newman.js 
+      env:
+        STRIPE_TEST_KEY: sk_test_7LEesZrYrgDtyxsc6Hy5i0lS00GVFEbmqb
+        ENDPOINT: http://localhost:5050
+
+    - name: Run tests against deployed preview
+      run:  cd tests && node newman.js 
       env:
-        CI: true
+        STRIPE_TEST_KEY: sk_test_7LEesZrYrgDtyxsc6Hy5i0lS00GVFEbmqb
+        ENDPOINT: https://deploy-preview-${{github.event.number}}--ambianic-serverless.netlify.app/.netlify/functions
+    
+    - name: Generate OpenAPI specs
```

Comment:
@vickywane this flow is backwards. Why is testing done BEFORE API is generated? That means the tests are run against an old version of the API, not the one changed in a new PR!

Reply:
This has been resolved. Generating the API is now the first step after setting up the environment